test/the-rewarder/README.md

```diff
@@ -0,0 +1,587 @@
+    ```solidity
+    console.log("GlobalStorage", address(glob));
+    console.log("DamnValuableToken", address(dvt));
+    console.log("WETH", address(weth));
+    console.log("TheRewarderDistributor", address(distributor));
+    ```
+    ```javascript
+    $ halmos --solver-timeout-assertion 0 --function check_theRewarder
+    ...
+    [console.log] GlobalStorage 0x00000000000000000000000000000000000000000000000000000000aaaa0002
+    [console.log] DamnValuableToken 0x00000000000000000000000000000000000000000000000000000000aaaa0003
+    [console.log] WETH 0x00000000000000000000000000000000000000000000000000000000aaaa0004
+    [console.log] TheRewarderDistributor 0x00000000000000000000000000000000000000000000000000000000aaaa0006
+    ```
+7. `vm.expectRevert()` is unsupportable cheat-code. Just delete it.
+### _isSolved() implementation
+The original checks look like:
+```solidity
+// Player saved as much funds as possible, perhaps leaving some dust
+assertLt(dvt.balanceOf(address(distributor)), 1e16, "Too much DVT in distributor");
+assertLt(weth.balanceOf(address(distributor)), 1e15, "Too much WETH in distributor");
+
+// All funds sent to the designated recovery account
+assertEq(
+    dvt.balanceOf(recovery),
+    TOTAL_DVT_DISTRIBUTION_AMOUNT - ALICE_DVT_CLAIM_AMOUNT - dvt.balanceOf(address(distributor)),
+    "Not enough DVT in recovery account"
+);
+assertEq(
+    weth.balanceOf(recovery),
+    TOTAL_WETH_DISTRIBUTION_AMOUNT - ALICE_WETH_CLAIM_AMOUNT - weth.balanceOf(address(distributor)),
+    "Not enough WETH in recovery account"
+);
+```
+As usual, we check whether we can empty the distributor of some unexpected amount. Note that the formula here will be a bit more complicated than we are used to seeing. First, we take into account that **Alice** has taken her reward and it is expected that the player himself can take his reward once. Therefore, the invariant looks like this:
+```solidity
+function _isSolved() private view {
+    assert (dvt.balanceOf(address(distributor)) >= 
+            TOTAL_DVT_DISTRIBUTION_AMOUNT - ALICE_DVT_CLAIM_AMOUNT - 11524763827831882);
+    assert (weth.balanceOf(address(distributor)) >= 
+            TOTAL_WETH_DISTRIBUTION_AMOUNT - ALICE_WETH_CLAIM_AMOUNT - 1171088749244340);
+}
+```
+`11524763827831882` and `1171088749244340` are the amounts of **DVT** and **WETH** the player is expected to be able to collect as he is one of the reward recipients. We took these numbers from **dvt-distribution.json** and **weth-distribution.json**.
+### Loading rewards
+In the setup process, the original test internally parses 1000 records in **JSON** format and uploads them to the **distrubutor** contract. However, there is a problem: Halmos does not support the required cheat codes, namely `vm.projectRoot()`, `vm.readFile()` and `vm.parseJson()`. We will work around this problem in a somewhat dirty but effective way. Instead of parsing the **JSON**, we will immediately explicitly insert the bytes into the right place. 
```

Comment:
```suggestion
In the setup process, the original test internally parses 1000 records in **JSON** format and uploads them to the **distributor** contract. However, there is a problem: Halmos does not support the required cheat codes, namely `vm.projectRoot()`, `vm.readFile()` and `vm.parseJson()`. We will work around this problem in a somewhat dirty but effective way. Instead of parsing the **JSON**, we will immediately explicitly insert the bytes into the right place. 
```